File: 10810-ultra.cpp
//UVa: 10810 - Ultra-QuickSort
#include <vector>
#include <iostream>
using namespace std;

vector<long long int> num;

long long int flip(int low, int hi){
	long long int r = 0;
	if(hi-low <= 1){
		return 0;
	}
	int mid = low + ((hi-low)/2);
	vector<long long int> aux;
	int a = low;
	int b = mid;

	r = flip(low,mid);
	r = r + flip(mid,hi);

	while(a<mid && b<hi){
		if(num[a] <= num[b]){
			aux.push_back(num[a]);
			a++;
		}else if(num[a]>num[b]){
			aux.push_back(num[b]);
			b++;
			r = r + (mid - a);
		}
	}
	while(a<mid){
		aux.push_back(num[a]);
		a++;
	}

	while(b<hi){
		aux.push_back(num[b]);
		b++;
	}
	int j = low;
	for(int i=0; i<aux.size(); i++){
		if(j<hi){
			num[j] = aux[i];
			j++;
		}
	}
	return r;
}

int main(){
	int n;
	long long int s,ans;
	cin >> n;
	while(n != 0){
		for(int i=0; i<n; i++){
			cin >> s;
			num.push_back(s);
		}
		ans = flip(0, n);
		cout << ans << endl;
		num.clear();
		cin >> n;
	}
}

File: 10810-ultra.py
#UVa: 10810 - Ultra-QuickSort
from sys import setrecursionlimit as SRL
from sys import stdin
SRL(10000)

def mergeSort(nums, lo, hi):
  ans = 0  
  if( hi-lo<=1 ):
    return 0
  mid = lo+((hi-lo) >> 1)
  aux,a,b = [],lo,mid

  ans = mergeSort(nums,lo,mid)
  ans += mergeSort(nums,mid,hi)

  while( a < mid and  b < hi):
    if( nums[a] <= nums[b]):
      aux.append(nums[a])
      a += 1
    elif( nums[a] > nums[b]):
      aux.append(nums[b])
      b += 1
      ans += mid - a
  
  while( a < mid):
    aux.append(nums[a])
    a += 1
  
  while( b < hi):
    aux.append(nums[b])
    b += 1

  nums[lo:hi] = aux
  return ans

def main():
  n = int(stdin.readline())
  while n:
    l = list()
    for e in range(n):
      l.append(int(stdin.readline()))
    print(mergeSort(l,0,n))
    n = int(stdin.readline())

main()